cpu/stm32l1/periph/pm.c

```diff
@@ -0,0 +1,160 @@
+/*
```

Comment:
A quick glimpse makes me feel that this could be integrated a lot more into the existing stm32 pm code.

Reply:
I actually thought the same and tried it but it looked really ugly. Lots of `#ifdef`'s everywhere. The architectures for power management between F and L are really different (registers, options, modes, etc.), so I'd say it deserves it's own implementation, even if it's similar.

Reply:
So, is it possible to have only two implementations then? One stm32f pm and one stm32l pm?

Reply:
haven't tested with other stm32l besides l1, I do have a nucleo-l0 somewhere, could test and see. But in that case what would be the suggested structure? ifdef's in stm32_common? It would be really ugly, or split stm32_common into a stm32l_common and stm32f_common? Somehting like that would need more opinions-ACK's

Reply:
L0-L1-L4 are pretty similar in power management (and other peripherals too). We have plans to port PM on L0 and L4, didn't make it yet, but I don't expect too many ifdefs there.

F-series CPUs are quite different, especially oldest F1 series.

P.S. Actually, even STM32 CPUs of the same series are not always fully compatible (e.g. STM32L151CB vs STM32L151CB-A vs STM32L151CC — RTC, EEPROM, some errata points are different), but we tested this LPM implementation on L151CB, L151CB-A, L151CC and L152RE, surprisingly it works.

Reply:
I can get working on that or merge efforts (@olegart depending on work-load), but I feel that would be a different PR.

Reply:
@fjmolinas yeah, for sure, that's for future PRs.

P.S. Actually I have a meeting with the client tomorrow evening, so I still have roughly a day to test and fix LPM on STM32L071KBU6 :)